internal/integrations/opaque/opaque.go

```diff
@@ -85,12 +85,9 @@ func (OpaqueIntegration) PrepareRun(ctx context.Context, server planning.Server,
 			return err
 		}
 
-		config := binary.Config
-		if config != nil {
-			run.WorkingDir = config.WorkingDir
-			run.Command = config.Command
-			run.Args = config.Args
-			run.Env = config.Env
+		err = run.BinaryConfig.MergeFrom(binary.Config)
+		if err != nil {
```

Comment:
Nit: please scope err, rather than re-using variables. It makes these types of declarations simpler to maintain:

```
if err := run.BinaryConfig.MergeFrom(...); err != nil {
```